src/Google.Cloud.Functions.Invoker/FunctionsStartup.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Google.Cloud.Functions.Invoker
+{
+    /// <summary>
+    /// Base class for startup classes providing additional configuration for functions
+    /// in the Functions Framework, typically configuring services and/or middleware.
+    /// These types are discovered using <see cref="FunctionsStartupAttribute"/>,
+    /// which also allows ordering to be specified.
```

Comment:
This is partially true only. We can ensure that services in the startups will be included in the service collection in that order but not middlewares (where the order is also important), because although we add the Startup classes to the services in the order indicated by the attribute, we have no guarante that when we get them on ConfigureApplication we'll get them on that order. I think you'll need a wrapper for the Startups that allows you to add an order both based on the attribute and on the actual order the single ones were included, so you can mimick that later on ConfigureApplication.
Or lets chat because it'll be easier to explain.

Reply:
Covered by earlier comment, I believe - I think the order *is* preserved. We could potentially expand the comment to talk about ordering with respect to explicitly-added startups, but it might add more noise than enlightenment.

Definitely happy to chat.